File: Javascript/WordPattern.js
var wordPattern = function(pattern,s){
    let sArray = s.split(" ")
    if(pattern.length !== sArray.length){
        return false
    }
    let hashMap = {}

    for(let i=0;i<pattern.length;i++){
        if(pattern[i] in hashMap){
            if(hashMap[pattern[i]] !== sArray[i]){
                return false
            }
        }
        else{
            const values = Object.values(hashMap)

            for(const v of values){
                if(v === sArray[i]){
                    return false
                }
            }

            hashMap[pattern[i]] = sList[i]
        }
    }

    return true

}

File: Python/WordPattern.py
def wordPattern(pattern,s):
    hashMap = {}
    sList = s.split(" ")

    if len(pattern) != len(sList):
        return False

    for i in range(len(pattern)):
        if pattern[i] not in hashMap:
            if sList[i] in hashMap.values():
                return False
            
            hashMap[pattern[i]] = sList[i]

        else:
            if hashMap[pattern[i]] != sList[i]:
                return False
    
    return True